draft-ietf-oauth-identity-chaining.md

```diff
@@ -366,20 +366,22 @@ The flow contains the following steps:
 
 Authorization servers may act as clients too. This can be necessary because of following reasons:
 
-* Resource servers may not have knowledge of authorization servers.
-* Resource servers may not have network access to other authorization servers.
-* A strict access control on resources outside the trust domain is required and enforced by authorization servers.
-* Authorization servers require client authentication. Managing clients for resource servers outside of the trust domain is not intended.
+* Clients in Domain A may not have knowledge of authorization servers in Domain B.
+* Clients in domain A may not have network access to other authorization servers in Domain B.
+* Strict access control on resources outside the trust domain is required and enforced by authorization servers in Domain B.
+* Authorization servers in domain B require client authentication, but are unable to manage clients outside of Domain B.
 
-Under these conditions, an authorization server in one domain (Domain A) may obtain an Access Token from an authorization server in another domain (Domain B) on-behalf-of the resource server in Domain A. This enables the resource server in Domain A to access a resource server in Domain B. The authorization server may use the flows described in this specification by acting as a client. The flow when authorization servers act as client would look like this:
+Under these conditions, an authorization server in one domain (Domain A) may obtain an Access Token from an authorization server in another domain (Domain B) on-behalf-of any client in Domain A. This enables clients in Domain A to access a protected resource server in Domain B. Resource servers in domain A may act as a client to the authorization server in Domain A in order to obtain an access token to access a protected resource in Domain B in order to complete a request.
+
+The authorization server may use the flows described in this specification by acting first as a client to itself to obtain an assertion grant and then act as a client to the authorization server in domain B to request an access token for protected resources in Domain B. The flow when authorization servers act as a client on-behalf of another client in it's own trust domain would look like this:
```

Comment:
I think we should on the terminology first before merging this.

Reply:
We could also change it later if we decide we want to change terminology - it is consistent with the current terminology and may help people better understand, parse the spec until such time as we decide to change terminology (or not).

@kburgin3 would love your thoughts on whether it is worth adding this clarification.

Reply:
> We could also change it later if we decide we want to change terminology - it is consistent with the current terminology and may help people better understand, parse the spec until such time as we decide to change terminology (or not).
> 
> @kburgin3 would love your thoughts on whether it is worth adding this clarification.

Same comment as before. Why not just say 

'...an authorization server in **Domain A** may obtain an Access Token from an authorization server in **Domain B**'

Reply:
Agreed - made the change. We use the "one domain"/"another domain" language throughout the doc. I created another issue to review usage and consider replacing them with Domain A/Domain B language.